### DISCUSSION TITLE
Make tar_watch aware of the nested lists in targets objects to allow for easy filtering

### DISCUSSION THREAD

**Discussion by brunocarlin** [ropensci/targets#300] [Ideas]
The concept is pretty simple right now when your pipeline gets big you get sort of a cluttered view of what is happening, one way to minimize this is using select filters, but another more interactive way would be to allow tar_watch to filter based on named lists and a sort of folder structure.

say we build this pipeline based on the targets-minimal repo

```
library(targets)
source("R/functions.R")
options(tidyverse.quiet = TRUE)
tar_option_set(packages = c("biglm", "tidyverse"))

get_data <- list(
  tar_target(
    raw_data_file,
    "data/raw_data.csv",
    format = "file"),
  tar_target(
    raw_data,
    read_csv(raw_data_file, col_types = cols())
  )
)


wrangle_data <-   list(tar_target(
    data,
    raw_data %>%
      mutate(Ozone = replace_na(Ozone, mean(Ozone, na.rm = TRUE)))
  )
)

inital_process <- list(get_data,wrangle_data)

post_process_data <- list(
  tar_target(hist, create_plot(data)),
  tar_target(fit, biglm(Ozone ~ Wind + Temp, data))
)


whole_pipeline <- list(inital_process,post_process_data)
```

Now this generated this amazing shiny page

![image](https://user-images.githubusercontent.com/1176377/107042603-b8ee8600-67a0-11eb-8ed7-45a3df57a01d.png)

If we could just get a selector based on objects it would already go a long way into quickly filtering the image, a nested selector would be even better

**Comment by wlandau** [comment]
Toggling on "targets only" is the easiest way to get a cleaner graph. Other kinds of target selection are trickier. For example, `targets` does not retain about the nesting in target lists. Another approach is to call `tar_manifest()` to get a list of target names and then use a multi-select widget to allow or exclude some, but that would decrease performance on refreshes.
